### PR TITLE
[MRG] FIX: orientation problem with plot_glass_brain

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Fixes issue #1815 

We have a work around to handle when the data is completely masked. This is to handle issues happened specific to matplotlib==2.1.0 as introduced in PR #1547.

But now it seems like it is not a great idea to use this work around for all versions of matplotlib as can be seen from issue #1815. This PR checks matplotlib version strictly to 2.1.0 as this issue is fixed in 2.1.1 or later.

### CODE REVIEW

**Comment by bthirion** [comment]
Well, I have this same issue with matplotlib 3.0.1.

**Comment by KamalakerDadi** [comment]
>Well, I have this same issue with matplotlib 3.0.1.

Sorry, cannot reproduce with this PR with matplotlib 3.0.1

**Comment by kchawla-pi** [comment]
Write a test for this change?

File: nilearn/plotting/displays.py [hunk 1]
```diff
@@ -791,8 +791,9 @@ def _map_show(self, img, type='imshow',
                 # If data_2d is completely masked, then there is nothing to
                 # plot. Hence, continued to loop over. This problem came up
                 # with matplotlib 2.1.0. See issue #9280 in matplotlib.
-                if data_2d.min() is np.ma.masked:
-                    continue
+                if LooseVersion(matplotlib.__version__) == LooseVersion("2.1.0"):
+                    if data_2d.min() is np.ma.masked:
+                        continue
```
Comment (kchawla-pi) [inline nilearn/plotting/displays.py]: DO we need to warn users when this happens?

**Comment by kchawla-pi** [comment]
> @kchawla-pi Can you check if this can be reproducible or cannot be reproducible with matplotlib 3.0.1

On it. Will do it once I have done my current PR.

**Comment by kchawla-pi** [comment]
nilearn-issue-1815-my-figure-0
This is what I got @KamalakerDadi

**Comment by KamalakerDadi** [comment]
I am looking forward for better ideas if this is not a right fix.

**Comment by kchawla-pi** [comment]
Hang on I ran the script with 0.5.0 instead of this PR. I'll run it again now.

**Comment by kchawla-pi** [comment]
nilearn-issue-1815-pr-figure-0
[nilearn-issue-1815-details.txt](https://github.com/nilearn/nilearn/files/2828375/nilearn-issue-1815-details.txt)
OKay all info atached

**Comment by kchawla-pi** [comment]
@KamalakerDadi Looks fixed to me. WHat do you mean not the right fix?
@bthirion can you send us the details of your environment where you can see the bug with MPL3.0.1?

**Comment by KamalakerDadi** [comment]
>nilearn-issue-1815-details.txt
OKay all info atached

Thanks a lot for the info. I would like to know if you think the BUG is fixed or not.

**Comment by KamalakerDadi** [comment]
>WHat do you mean not the right fix?

I don't know. Whatever you think about this PR.

**Comment by kchawla-pi** [comment]
[nilearn-issue-1815-details-mpl210.txt](https://github.com/nilearn/nilearn/files/2828488/nilearn-issue-1815-details-mpl210.txt)
nilearn-issue-1815-pr-mpl210-figure-0


It doesn't seem fixed...

**Comment by KamalakerDadi** [comment]
Yeah. Good point.

**Comment by bthirion** [comment]
No that's fine, I obtain a correct display. Sorry for the noise.

**Comment by KamalakerDadi** [comment]
>It doesn't seem fixed...

@kchawla-pi ready for another round.

**Comment by KamalakerDadi** [comment]
>    It doesn't seem fixed...

>@kchawla-pi ready for another round.

Nope, some of the examples are not happy with recent work around.

**Comment by KamalakerDadi** [comment]
Also, this PR fixes the glass brain orientation when direction 'l' is given with stat_map_img=None.
 ```python
plotting.plot_glass_brain(None, display_mode='lr')
```

**Comment by KamalakerDadi** [comment]
Code to test this PR.

```python
import matplotlib
print(matplotlib.__version__)
from nilearn import plotting
from nilearn import datasets
from nilearn.image import new_img_like

ho = datasets.fetch_atlas_harvard_oxford('sub-maxprob-thr25-2mm',
                                         symmetric_split=True)
data = ho.maps.get_data()
this_label = (data == 19)
img = new_img_like(ho.maps, this_label)
display = plotting.plot_glass_brain(img, display_mode='lyrz')
display = plotting.plot_glass_brain(img, display_mode='l')
display = plotting.plot_glass_brain(img, display_mode='r')
display = plotting.plot_glass_brain(img, display_mode='lzr')
display = plotting.plot_glass_brain(img, display_mode='lyr')
display = plotting.plot_glass_brain(img, display_mode='lzry')
plotting.show()
```

**Comment by bthirion** [comment]
Fine on my box.

**Comment by KamalakerDadi** [comment]
>Fine on my box.

Thanks. 

I am almost there. I am testing with ```display.add_overlay```. If this works nothing to do except adding some tests and ```doc/whats_new.rst```.

**Comment by KamalakerDadi** [comment]
@kchawla-pi you have permissions to rebuild AppVeyor ?

**Comment by kchawla-pi** [comment]
Yes. I did rebuild it for this one.

**Comment by kchawla-pi** [comment]
Its stuck on Queued, I'll cancel it and rerun.

**Comment by KamalakerDadi** [comment]
This one is now ready for one more review.

**Comment by KamalakerDadi** [comment]
This PR needs atleast +2 to merge. To double check that nothing has broken with this fix and as it is tricky.

**Comment by KamalakerDadi** [comment]
@chrplr if you could review this and tell us if it fixes the problem. That would be great.

**Comment by kchawla-pi** [comment]
Don't forget the what's new if it's not there yet.

**Comment by KamalakerDadi** [comment]
If somebody could quickly check the example gallery after the documentation is build. That would be great.

**Comment by GaelVaroquaux** [comment]
Given the feedback from @chrplr this is a :+1: for mrg.

**Comment by kchawla-pi** [comment]
Hey @KamalakerDadi should I resolve the conflict and merge this?

**Comment by jeromedockes** [comment]
I think it would be good to merge this one before the release?

**Comment by kchawla-pi** [comment]
OK I'll do that now.

**Pull request merged**